lib/friendly_shipping/services/ship_engine/parse_label_response.rb

```diff
@@ -17,21 +17,23 @@ def self.call(request:, response:)
           else
             label_url = label_uri_string
           end
-          [
-            FriendlyShipping::Label.new(
-              id: parsed_json['label_id'],
-              shipment_id: parsed_json['shipment_id'],
-              tracking_number: parsed_json['tracking_number'],
-              service_code: parsed_json['service_code'],
-              label_href: label_url,
-              label_data: label_data,
-              label_format: parsed_json['label_format'].to_sym,
-              shipment_cost: parsed_json['shipment_cost']['amount'],
-              data: parsed_json,
-              original_request: request,
-              original_response: response
-            )
-          ]
+          ApiResult.new(
+            [
+              FriendlyShipping::Label.new(
+                id: parsed_json['label_id'],
+                shipment_id: parsed_json['shipment_id'],
+                tracking_number: parsed_json['tracking_number'],
+                service_code: parsed_json['service_code'],
+                label_href: label_url,
+                label_data: label_data,
+                label_format: parsed_json['label_format'].to_sym,
+                shipment_cost: parsed_json['shipment_cost']['amount'],
+                data: parsed_json
+              )
```

Comment:
Non-blocking nit, but I think this would be slightly more readable if it were assigned to a `label` variable and then that `label` variable was referenced when constructing the `ApiResult`.

Reply:
Yes, that's true.

Reply:
Fixed